src/sentry/testutils/factories.py

```diff
@@ -1359,7 +1367,7 @@ def create_userreport(project, event_id=None, **kwargs):
         )
 
         return UserReport.objects.create(
-            group_id=event.group.id,
+            group_id=event.group_id,
```

Comment:
There was some typing issue here where `event.group` is unclear to be either `Group` or `None` since `event` could either be `Event` or `GroupEvent`.

I tried fixing it, however, it is easier to just use `group_id` which is shared by both classes.
https://github.com/getsentry/sentry/blob/932aa3e8f6f80f7791542af0965badf70a7a3c12/src/sentry/eventstore/models.py#L629
https://github.com/getsentry/sentry/blob/932aa3e8f6f80f7791542af0965badf70a7a3c12/src/sentry/eventstore/models.py#L734

One day I will try to fix it but right now it's been taking too much of my time.

Reply:
We can probably simplify `GroupEvent` a lot back into `Event`. It was mostly a thing from when we first introduced performance issues, and an event could have multiple groups associated with it. Now the main thing that differentiates a `GroupEvent` is having an `occurrence`, so there's probably a reasonable amount of consolidation we can do.